homeassistant/components/notify/kodi.py

```diff
@@ -0,0 +1,73 @@
+﻿"""
+Kodi notification service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.kodi/
+"""
+import logging
+
+from homeassistant.components.notify import (ATTR_TITLE, ATTR_TITLE_DEFAULT,
+                                             ATTR_DATA,
+                                             BaseNotificationService)
+
+_LOGGER = logging.getLogger(__name__)
+REQUIREMENTS = ['jsonrpc-requests==0.3']
+
+ATTR_DISPLAYTIME = 'displaytime'
+ATTR_ICON = 'icon'
+
+
+def get_service(hass, config):
+    """Return the notify service."""
+    url = '{}:{}'.format(config.get('host'), config.get('port', '8080'))
+
+    jsonrpc_url = config.get('url')  # deprecated
+    if jsonrpc_url:
+        url = jsonrpc_url.rstrip('/jsonrpc')
+
+    auth = (config.get('username', ''),
+            config.get('password', ''))
+
+    return KODINotificationService(
+        config.get('name', 'Kodi'),
+        url,
+        auth
+    )
+
+
+# pylint: disable=too-few-public-methods
+class KODINotificationService(BaseNotificationService):
+    """Implement the notification service for Kodi."""
+    def __init__(self, name, url, auth=None):
+        """Initialize the service."""
+        import jsonrpc_requests
+        self._name = name
+        self._url = url
+        self._server = jsonrpc_requests.Server(
+            '{}/jsonrpc'.format(self._url),
+            auth=auth,
+            timeout=5)
+
+    def send_message(self, message="", **kwargs):
+        """Send a message to Kodi."""
+        import jsonrpc_requests
+        try:
+            data = kwargs.get(ATTR_DATA)

```

Comment:
Here you can ensure data is not `None`. With the second it will also catch if you pass in None `send_message(data=None)`
```python
data = kwargs.get(ATTR_DATA, {})
-or-
data = kwargs.get(ATTR_DATA) or {}
```